### PR TITLE
Feature game factory

### PR DESCRIPTION
**Pull request by MatthewIsayan**
- `setup()` responsible for initial game modifications before execution of game loop
- `setup()` could be moved to `GameState` if only `GameState` requires initial modifications
- Abstract `GameFactory` that creates and sets up a game ready to run

### CODE REVIEW

File: src/tilematch_tools/core/game_factory.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        LOGGER.warning('Using default implementation in setup(). This is meant to be overridden!')
+        pass
+            
+
+class GameFactory(ABC):
+    """
+        Responsible for creating a game ready to run.
+    """
+    @abstractmethod
+    def create_game(state: GameState, loop : GameLoop, view: GameView, tick_speed: int):
```
Comment (nathancmendoza) [inline src/tilematch_tools/core/game_factory.py:30]: If this does not operate on an instance, decorate with `@staticmethod` too.

**Comment by MatthewIsayan** [comment]
- Removed `@abstractmethod` from `setup()`, because not all games necessarily have to implement this if they don't need a setup
- Assuming we want board dimensions and `tick_speed` variable, I've included them as parameters for the execution environment to control
```
def create_game(row, cols, tick_speed):
            # Example snippet   
            board = BoardFactory.create_board(ColumnsBoard, rows, cols)
            score = ColumnsScoring()
            state = ColumnsGameState(board, score)
            game =  ColumnsGame(state, ColumnsGameLoop, ColumnsView, tick_speed)
            game.setup()
            return game
```

**Comment by MatthewIsayan** [comment]
> Assuming we want board dimensions and tick_speed variable, I've included them as parameters for the execution environment to control
- Removed these params
- Updated snippet
```
def create_game():
            # Example snippet   
            board = BoardFactory.create_board(ColumnsBoard, 10, 10)
            score = ColumnsScoring()
            state = ColumnsGameState(board, score)
            game =  ColumnsGame(state, ColumnsGameLoop, ColumnsView, 750_000_000)
            game.setup()
            return game
```

**Review by nathancmendoza — APPROVED** [review]
Nice.
I see you've discovered the `Mock` objects. Truly wonderful things, but something that needs extra care to use.

**Pull request merged**